File: python/q34.py
#!/usr/bin/python3

import math

def factSum(n):
    runningTotal = 0
    for digit in str(n):
        runningTotal += math.factorial(int(digit))
    return runningTotal


#I don't know an upper bound so just going to guess 100,000 and hope the program doesn't take too long
total = 0
for i in range(3,100000):
    if factSum(i)==i:
        total += i
print(total)

File: go/q34.go
package main

import (
	"fmt"
	"github.com/daniel-willett/common-go-methods"
	"strconv"
)

func factSum(n int) int{
	var runningTotal int = 0
	var stringN string = strconv.Itoa(n)
	for _, digit := range stringN{
		temp, _ := common.Factorial(int(digit)-48)
		runningTotal += temp
	}
	return runningTotal
}

func main(){
	var total int = 0
	for i:=3; i<100000; i++{
		if factSum(i)==i{
			total += i
		}
	}
	fmt.Println(total)
}
